GVRf/Framework/jni/objects/hybrid_object_jni.cpp

```diff
@@ -28,7 +28,9 @@ Java_org_gearvrf_NativeHybridObject_delete(JNIEnv * env,
 JNIEXPORT void JNICALL
 Java_org_gearvrf_NativeHybridObject_delete(JNIEnv * env,
         jobject obj, jlong jhybrid_object) {
-    delete reinterpret_cast<HybridObject*>(jhybrid_object);
+    HybridObject* hybrid_object = reinterpret_cast<HybridObject*>(jhybrid_object);
+    delete hybrid_object;
+    hybrid_object = nullptr;
```

Comment:
So, you guys are still working on this, right? I am seeing a SIGSEGV crash:

07-07 12:46:10.770 27961 28013 F libc    : Fatal signal 11 (SIGSEGV) at 0x00000010 (code=1), thread 28013 (GVRF Finalize T)
07-07 12:46:10.800 27961 28041 W TimeWarp: V-sync 5756: Eye 1, CPU latency 0.007, GPU latency 0.002, Total latency = 0.009
07-07 12:46:10.830   317   317 I DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
07-07 12:46:10.830   317   317 I DEBUG   : Build fingerprint: 'samsung/trltetmo/trltetmo:4.4.4/KTU84P/N910TUVU1ANK4:user/release-keys'
07-07 12:46:10.830   317   317 I DEBUG   : Revision: '12'
07-07 12:46:10.830   317   317 I DEBUG   : pid: 27961, tid: 28013, name: GVRF Finalize T  >>> org.gearvrf.gvroutlinesample <<<
07-07 12:46:10.830   317   317 I DEBUG   : signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 00000010
…
07-07 12:46:10.980   317   317 I DEBUG   : backtrace:
07-07 12:46:10.980   317   317 I DEBUG   :     #00  pc 00000010  <unknown>
07-07 12:46:10.980   317   317 I DEBUG   :     #01  pc 000c4b40  /data/app-lib/org.gearvrf.gvroutlinesample-15/libgvrf.so (Java_org_gearvrf_NativeHybridObject_delete+20)
07-07 12:46:10.980   317   317 I DEBUG   :     #02  pc 000210cc  /system/lib/libdvm.so (dvmPlatformInvoke+112)
07-07 12:46:10.980   317   317 I DEBUG   :     #03  pc 00051daf  /system/lib/libdvm.so (dvmCallJNIMethod(unsigned int const*, JValue*, Method const*, Thread*)+398)
…


From: Phil Lira [mailto:notifications@github.com]
Sent: Tuesday, July 07, 2015 5:52 AM
To: Samsung/GearVRf
Cc: Jason Kim
Subject: Re: [GearVRf] Added Support to Multipass rendering and Created a Sample to showcase outline rendering. (#143)


In GVRf/Framework/jni/objects/hybrid_object_jni.cpp<https://github.com/Samsung/GearVRf/pull/143#discussion_r34035433>:

> @@ -28,7 +28,9 @@ Java_org_gearvrf_NativeHybridObject_delete(JNIEnv * env,

>  JNIEXPORT void JNICALL

>  Java_org_gearvrf_NativeHybridObject_delete(JNIEnv * env,

>          jobject obj, jlong jhybrid_object) {

> -    delete reinterpret_cast<HybridObject*>(jhybrid_object);

> +    HybridObject* hybrid_object = reinterpret_cast<HybridObject*>(jhybrid_object);

> +    delete hybrid_object;

> +    hybrid_object = nullptr;

Thanks Chao. I commited the updated changes to fix it plus reverted hybrid_object_jni.cpp.

I still have some questions.

1) How should we set native pointer to null to avoid native code from mistakenly referencing deleted pointers? Cause I think right now, doing checks for nullptr in native code are non effective.

2) I'm a little confused with GVRHybridObject. There's a mNativePointer on GVRHybridObject and another copy of it in GVRReference. The GVRHybridObject initialized both references upon construction.

mNativePointer = nativePointer;
sReferenceSet.add(new GVRReference(this, nativePointer, cleanupHandlers));

When closing an object the GVRReference native pointer is deleted and set to null but shouldn't GVRHybridObject native pointer also be set to null?

—
Reply to this email directly or view it on GitHub<https://github.com/Samsung/GearVRf/pull/143/files#r34035433>.